accounts/abi/bind/testdata/v2/nested_libraries/bindings.go

```diff
@@ -0,0 +1,442 @@
+	out, err := _L4.abi.Unpack("Do", data)
+
+	if err != nil {
+		return *new(*big.Int), err
+	}
+
+	out0 := *abi.ConvertType(out[0], new(*big.Int)).(**big.Int)
+
+	return out0, err
+
+}
+
+var L4bLibraryDeps = []*bind.MetaData{}
+
+// TODO: convert this type to value type after everything works.
+// L4bMetaData contains all meta data concerning the L4b contract.
+var L4bMetaData = &bind.MetaData{
+	ABI:     "[{\"inputs\":[{\"internalType\":\"uint256\",\"name\":\"val\",\"type\":\"uint256\"}],\"name\":\"Do\",\"outputs\":[{\"internalType\":\"uint256\",\"name\":\"\",\"type\":\"uint256\"}],\"stateMutability\":\"pure\",\"type\":\"function\"}]",
+	Pattern: "6070639404c39b5667691bb1f9177e1eac",
+	Bin:     "0x61025161004d600b8282823980515f1a6073146041577f4e487b71000000000000000000000000000000000000000000000000000000005f525f60045260245ffd5b305f52607381538281f3fe7300000000000000000000000000000000000000003014608060405260043610610034575f3560e01c80632ad1127214610038575b5f80fd5b610052600480360381019061004d9190610129565b610068565b60405161005f9190610163565b60405180910390f35b5f600173__$fd1474cf57f7ed48491e8bfdfd0d172adf$__632ad11272846040518263ffffffff1660e01b81526004016100a29190610163565b602060405180830381865af41580156100bd573d5f803e3d5ffd5b505050506040513d601f19601f820116820180604052508101906100e19190610190565b6100eb91906101e8565b9050919050565b5f80fd5b5f819050919050565b610108816100f6565b8114610112575f80fd5b50565b5f81359050610123816100ff565b92915050565b5f6020828403121561013e5761013d6100f2565b5b5f61014b84828501610115565b91505092915050565b61015d816100f6565b82525050565b5f6020820190506101765f830184610154565b92915050565b5f8151905061018a816100ff565b92915050565b5f602082840312156101a5576101a46100f2565b5b5f6101b28482850161017c565b91505092915050565b7f4e487b71000000000000000000000000000000000000000000000000000000005f52601160045260245ffd5b5f6101f2826100f6565b91506101fd836100f6565b9250828201905080821115610215576102146101bb565b5b9291505056fea2646970667358221220819bc379f2acc661e3dba3915bee83164e666ab39a92d0bcbf56b2438c35f2e164736f6c634300081a0033",
+}
+
+// L4b is an auto generated Go binding around an Ethereum contract.
+type L4b struct {
+	abi abi.ABI
+}
+
+// NewL4b creates a new instance of L4b.
+func NewL4b() (*L4b, error) {
+	parsed, err := L4bMetaData.GetAbi()
+	if err != nil {
+		return nil, err
+	}
+	return &L4b{abi: *parsed}, nil
+}
+
+// TODO: create custom exported types where unpack would generate a struct return.
+
+// TODO: test constructor with inputs
+func (_L4b *L4b) PackConstructor() ([]byte, error) {
+	return _L4b.abi.Pack("")
+}
+
+// Do is a free data retrieval call binding the contract method 0x2ad11272.
+//
+// Solidity: function Do(uint256 val) pure returns(uint256)
+func (_L4b *L4b) PackDo(val *big.Int) ([]byte, error) {
+	return _L4b.abi.Pack("Do", val)
+}
+
+func (_L4b *L4b) UnpackDo(data []byte) (*big.Int, error) {
+	out, err := _L4b.abi.Unpack("Do", data)
+
+	if err != nil {
+		return *new(*big.Int), err
```

Comment:
this looks funky. What's wrong with `return nil, err`? :)

Reply:
I think this is a result of the template being written to handle returning both pointer/value types without having a branch here.  IMO, worth improving this to make the generated code cleaner for pointer types:

if pointer type:  `return nil, err`.  if value type, `return {{defaultInstanceOfType}}, err`.

It also made me realize that my handling of returns for errors assumes pointer type return, so that will need to change.  And we will want to have contracts that expose this detail included so that we get test coverage via the bindings tests.